src/Confluent.Kafka/IProducer.cs

```diff
@@ -339,6 +339,14 @@ public interface IProducer<TKey, TValue> : IClient
         ///     Any produce call outside an on-going transaction, or for a failed
         ///     transaction, will fail.
         /// </remark>
+        /// <exception cref="KafkaTxnRequiresAbortException">
```

Comment:
can librdkafka cause this to occur here?

Reply:
Nope, see https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L6699